### PR TITLE
Port ec2_vpc_net to boto3 and add support to expand existing VPCs - fixes #31216

### LINKED ISSUE #31216: support adding secondary CIDRs to existing VPCs
Allow the addition of secondary IPv4 CIDRS to existing VPCs.

https://aws.amazon.com/about-aws/whats-new/2017/08/amazon-virtual-private-cloud-vpc-now-allows-customers-to-expand-their-existing-vpcs/

##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
lib/ansible/modules/cloud/amazon

##### ANSIBLE VERSION
```
2.5.0
```

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Fixes #31216.

~WIP until tests in #33111 are merged.~

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/amazon/ec2_vpc_net.py

##### ANSIBLE VERSION
```
2.5.0
```

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/46529/47/tests

**Comment by gundalow** [comment]
As agreed in 2017-12-07 Core meeting, we've done bulk pep8 fixes across all modules in `devel`.

This was done in https://github.com/ansible/ansible/pull/33682 and https://github.com/ansible/ansible/pull/33690 which have been merged into `devel`. because of this you will need to rebase your PR. Thanks in advance.

File: lib/ansible/modules/cloud/amazon/ec2_vpc_net.py [hunk 1]
```diff
@@ -16,61 +16,61 @@
 module: ec2_vpc_net
 short_description: Configure AWS virtual private clouds
 description:
-    - Create or terminate AWS virtual private clouds.  This module has a dependency on python-boto.
+    - Create, modify, and terminate AWS virtual private clouds.
 version_added: "2.0"
 author: Jonathan Davila (@defionscode)
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_net.py]: This is missing your authorship @s-hertel - a wide-ranging rewrite deserves attribution

File: lib/ansible/modules/cloud/amazon/ec2_vpc_net.py [hunk 2]
```diff
@@ -136,198 +136,206 @@
...
-            # get the vpc obj again in case it has changed
-            try:
-                vpc_obj = connection.get_all_vpcs(vpc_obj.id)[0]
-            except BotoServerError as e:
-                e_msg=boto_exception(e)
-                module.fail_json(msg=e_msg)
+                try:
+                    connection.modify_vpc_attribute(VpcId=vpc_id, EnableDnsSupport={'Value': dns_support})
+                except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                    module.fail_json_aws(e, "Failed to update enabled dns support attribute")
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_net.py]: Does the second parameter need `msg=` or does it just work ok without?

Comment (s-hertel) [reply lib/ansible/modules/cloud/amazon/ec2_vpc_net.py]: Tested to be sure, and it works without.

**Comment by s-hertel** [comment]
Rebased and removed pep8 changes.

**Comment by willthames** [comment]
Great job @s-hertel, thanks!

**Comment by brandond** [comment]
FYI, this PR broke tags.

**Pull request merged**